src/firestore/firekit.ts

```diff
@@ -773,6 +794,115 @@ export class RoarFirekit {
+        const age: number = Number(userData.age_year);
+        const yearOffset = Math.floor(age);
+        const monthOffset = (age % yearOffset) * 10;
+        let calcDob = new Date();
+        calcDob.setFullYear(calcDob.getFullYear() - yearOffset);
+        calcDob.setMonth(calcDob.getMonth() - monthOffset);
+        _set(userObject, 'studentData.dob', calcDob);
+      }
+      if(_get(userData, 'age_month')){
+        const age: number = Number(userData.age_year);
+        const monthOffset = Math.floor(age);
+        let calcDob = new Date();
+        calcDob.setMonth(calcDob.getMonth() - monthOffset);
+        _set(userObject, 'studentData.dob', calcDob);
+      }
+      if(_get(userData, 'dob')) _set(userObject, 'studentData.dob', userData.dob);
+      if(_get(userData, 'gender')) _set(userObject, 'studentData.gender', userData.gender);
+      if(_get(userData, 'ell_status')) _set(userObject, 'studentData.ell_status', userData.ell_status);
+      if(_get(userData, 'iep_status')) _set(userObject, 'studentData.iep_status', userData.iep_status)
+      if(_get(userData, 'frl_status')) _set(userObject, 'studentData.frl_status', userData.frl_status);
+
+      const dateNow = Date.now()
+      // create district entry
+      const districtId = _get(userData, 'district');
+      if(districtId) {
+        _set(userObject, 'districts', {
+          current: [districtId],
+          all: [districtId],
+          dates: {
+            [districtId!]: {
+              from: dateNow,
+              to: null
+            }
+          }
+        })
+      }
+      // create school entry
+      const schoolId = _get(userData, 'school');
+      if(schoolId){
+        _set(userObject, 'schools', {
+          current: [schoolId],
+          all: [schoolId],
+          dates: {
+            [schoolId!]: {
+              from: dateNow,
+              to: null
+            }
+          }
+        })
+      }
+      // create class entry
+      const classId = _get(userData, 'class');
+      if(classId){
+        _set(userObject, 'classes', {
+          current: [classId],
+          all: [classId],
+          dates: {
+            [classId!]: {
+              from: dateNow,
+              to: null
+            }
+          }
+        })
+      }
+      const cloudCreateAdminStudent = httpsCallable(this.admin.functions, 'createstudent');
+      const adminResponse = await cloudCreateAdminStudent({email, password, userData});
+      const adminUid = _get(adminResponse, 'data.adminUid');
+
+      const cloudCreateAppStudent = httpsCallable(this.app.functions, 'createstudent');
+      const appResponse = await cloudCreateAppStudent({adminUid, email, password, userData});
+      // cloud function returns all relevant Uids (since at this point, all of the associations and claims have been made)
+      return _get(appResponse, 'data.assessmentUid');
+      // Note: The assessment createstudent cloud function handles setting up the user claim.
```

Comment:
In keeping with my requested changes on your roar-firebase-functions PR, I think that we need to call that final cloud function here.

```suggestion
      // Note: The assessment createstudent cloud function handles setting up the user claim.
      
      const associateUserClaims = httpsCallable(this.admin.functions, 'associateuserclaims');
      const claimsResponse = await associateUserClaims({ adminUid, assessmentUid });
```